src/vm/finalizerthread.cpp

```diff
@@ -22,13 +22,22 @@ BOOL FinalizerThread::fQuitFinalizer = FALSE;
 
 #if defined(__linux__) && defined(FEATURE_EVENT_TRACE)
 #define LINUX_HEAP_DUMP_TIME_OUT 10000
+#define PERFORM_TASKS_IN_FINALIZER_WAIT_TIMEOUT
 
 extern bool s_forcedGCInProgress;
 ULONGLONG FinalizerThread::LastHeapDumpTime = 0;
 
 Volatile<BOOL> g_TriggerHeapDump = FALSE;
 #endif // __linux__
 
+
+#ifdef FEATURE_UNIX_LOW_MEMORY_NOTIFICATION
```

Comment:
I would not create a special feature for that. The features are good for large scale changes, but we want this to be always enabled.